pkg/federation/sdlmerge/remove_scalar_duplicates.go

```diff
@@ -0,0 +1,52 @@
+package sdlmerge
+
+import (
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astvisitor"
+)
+
+type removeDuplicateScalarTypeDefinitionVisitor struct {
+	operation     *ast.Document
+	scalarSet     map[string]bool
+	nodesToRemove []ast.Node
+	lastRef       int
+}
+
+func newRemoveDuplicateScalarTypeDefinitionVistior() *removeDuplicateScalarTypeDefinitionVisitor {
+	return &removeDuplicateScalarTypeDefinitionVisitor{
+		nil,
+		make(map[string]bool),
+		make([]ast.Node, 0),
+		-1,
+	}
+}
+
+func (r *removeDuplicateScalarTypeDefinitionVisitor) EnterDocument(operation, definition *ast.Document) {
+	r.operation = operation
```

Comment:
I wouldn't call it `r.operation` because we are not working on an operation here.
I know that the document is passed through `operation` in `EnterDocument` but this is because of the generic nature of the visitor.

`r.document` or `r.definition` could be better